dask/typing.py

```diff
@@ -0,0 +1,372 @@
+        final in-memory array.
+
+        Returns
+        -------
+        PostComputeCallable
+            Callable that recieves the sequence of the results of each
+            final key along with optional arguments. The signure must
+            be ``finalize(results: Sequence[Any], *args)``.
+        tuple[Any, ...]
+            Optional arguments passed to the function following the
+            key results (the `*args` part of the
+            ``PostComputeCallable``. If no additional arguments are to
+            be passed then this must be an empty tuple.
+
+        """
+
+    def __dask_postpersist__(self) -> tuple[PostPersistCallable, tuple]:
+        """Rebuilder function and optional arguments to contruct a persisted collection.
+
+        Returns
+        -------
+        PostPersistCallable
+            Callable that rebuilds the collection. The signature
+            should be
+            ``rebuild(dsk: Mapping, *args: Any, rename: Mapping[str, str] | None)``.
+            The callable should return an equivalent Dask collection
+            with the same keys as `self`, but with results that are
+            computed through a different graph. In the case of
+            :py:func:`dask.persist`, the new graph will have just the
+            output keys and the values already computed.
+        tuple[Any, ...]
+            Optional arugments passed to the rebuild callable. If no
+            additional arguments are to be passed then this must be an
+            empty tuple.
+
+        """
+
+    def __dask_tokenize__(self) -> Hashable:
+        """Value that must fully represent the object."""
+
+    @staticmethod
+    def __dask_optimize__(
+        dsk: MutableMapping,
+        keys: Sequence[Hashable],
+        **kwargs: Any,
+    ) -> MutableMapping:
+        """Given a graph and keys, return a new optimized graph.
+
+        This method can be either a ``staticmethod`` or a
+        ``classmethod``, but not an ``instancemethod``.
+
+        Note that graphs and keys are merged before calling
+        ``__dask_optimize__``; as such, the graph and keys passed to
+        this method may represent more than one collection sharing the
+        same optimize method.
+
+        Parameters
+        ----------
+        dsk : MutableMapping
+            The merged graphs from all collections sharing the same
+            ``__dask_optimize__`` method.
+        keys : Sequence[Hashable]
+            A list of the outputs from ``__dask_keys__`` from all
+            collections sharing the same ``__dask_optimize__`` method.
+        **kwargs : Any
+            Extra keyword arguments forwarded from the call to
+           ``compute`` or ``persist``. Can be used or ignored as
+           needed.
+
+        Returns
+        -------
+        MutableMapping
+            The optimized Dask graph.
+
+        """
+
+    __dask_scheduler__: staticmethod[SchedulerStaticMethod]
```

Comment:
This was an oversight on my part, it was a temporary experiment I meant to come back to while working on the `SchedulerStaticMethod` protocol, I forgot to come back to it because it doesn't actually cause a type error, which is kind of interesting. I'm running into some issues with both this and `__dask_optimize__`, I'll summarize in a non-inline comment